libbpfgo/libbpfgo.go

```diff
@@ -302,6 +302,40 @@ func (m *Module) GetMap(mapName string) (*BPFMap, error) {
 	}, nil
 }
 
+func (b *BPFMap) GetValue(key interface{}, valueSize int) ([]byte, error) {
+	var keyPtr unsafe.Pointer
+	if k, isType := key.(int8); isType {
+		keyPtr = unsafe.Pointer(&k)
+	} else if k, isType := key.(uint8); isType {
+		keyPtr = unsafe.Pointer(&k)
+	} else if k, isType := key.(int32); isType {
+		keyPtr = unsafe.Pointer(&k)
+	} else if k, isType := key.(uint32); isType {
+		keyPtr = unsafe.Pointer(&k)
+	} else if k, isType := key.(int64); isType {
+		keyPtr = unsafe.Pointer(&k)
+	} else if k, isType := key.(uint64); isType {
+		keyPtr = unsafe.Pointer(&k)
+	} else if k, isType := key.([]byte); isType {
+		keyPtr = unsafe.Pointer(&k[0])
+	} else {
+		return nil, fmt.Errorf("failed to update map %s: unknown key type %T", b.name, key)
+	}
+
+	// Create pointer to value byte array
+	if valueSize == 0 {
+		return nil, fmt.Errorf("valueSize must be > 0")
+	}
+	value := make([]byte, valueSize)
```

Comment:
So firstly, again very happy to follow to and the core maintainer's lead in what you want your project to look like. But to me, a wrapper/healper library should stop you from making _invalid_ decisions, but not _bad_ ones.

If someone wanted to use libbpfgo to create a map with an unreasonable key or value size, that's on them - In the kernel source, [ArrayMap](https://github.com/torvalds/linux/blob/d635a69dd4981cc51f90293f5f64268620ed1565/kernel/bpf/arraymap.c#L72) values for example just have to be less than `KMALLOC_MAX_SIZE`, which is pretty big.

It would make sense for us to restrict the size to `KMALLOC_MAX_SIZE`, but I feel that that setting the limit to anything smaller (which might be more reasonable to you and me), is being more opinionated than a library should. 

Looking at the [libbpf source](https://github.com/libbpf/libbpf/blob/42baefba714b2cce03fbfd68ecfff507a80adc1d/src/bpf.c#L80), they impose no restriction on value or key size, only passing the values on to the kernel to deal with